internal/reader/rmn_home.go

```diff
@@ -0,0 +1,311 @@
+		secondaryConfigDigest = versionedConfigWithDigests[1].ConfigDigest
+	}
+
+	r.setRMNHomeState(
+		primaryConfigDigest,
+		secondaryConfigDigest,
+		convertOnChainConfigToRMNHomeChainConfig(r.lggr, versionedConfigWithDigests))
+
+	if len(versionedConfigWithDigests) == 0 {
+		// That's a legitimate case if there are no rmn configs on chain yet
+		r.lggr.Warnw("no on chain configs found")
+		return nil
+	}
+
+	return nil
+}
+
+func (r *RmnHomePoller) setRMNHomeState(
+	primaryConfigDigest cciptypes.Bytes32,
+	secondaryConfigDigest cciptypes.Bytes32,
+	rmnHomeConfig map[cciptypes.Bytes32]rmntypes.RMNHomeConfig) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	s := &r.rmnHomeState
+
+	s.primaryConfigDigest = primaryConfigDigest
+	s.secondaryConfigDigest = secondaryConfigDigest
+	s.rmnHomeConfig = rmnHomeConfig
+}
+
+func (r *RmnHomePoller) GetRMNNodesInfo(configDigest cciptypes.Bytes32) ([]rmntypes.RMNHomeNodeInfo, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.rmnHomeState.rmnHomeConfig[configDigest].Nodes, nil
+}
+
+func (r *RmnHomePoller) IsRMNHomeConfigDigestSet(configDigest cciptypes.Bytes32) (bool, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	_, ok := r.rmnHomeState.rmnHomeConfig[configDigest]
+	return ok, nil
+}
+
+func (r *RmnHomePoller) GetMinObservers(configDigest cciptypes.Bytes32) (map[cciptypes.ChainSelector]uint64, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.rmnHomeState.rmnHomeConfig[configDigest].MinObservers, nil
+}
+
+func (r *RmnHomePoller) GetOffChainConfig(configDigest cciptypes.Bytes32) (cciptypes.Bytes, error) {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.rmnHomeState.rmnHomeConfig[configDigest].OffchainConfig, nil
+}
+
+func (r *RmnHomePoller) Close() error {
+	return r.sync.StopOnce(r.Name(), func() error {
+		defer r.wg.Wait()
+		close(r.stopCh)
+		return nil
+	})
+}
+
+func (r *RmnHomePoller) Ready() error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.sync.Ready()
+}
+
+func (r *RmnHomePoller) HealthReport() map[string]error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	if r.failedPolls >= MaxFailedPolls {
+		r.sync.SvcErrBuffer.Append(fmt.Errorf("polling failed %d times in a row", MaxFailedPolls))
+	}
+	return map[string]error{r.Name(): r.sync.Healthy()}
+}
+
+func (r *RmnHomePoller) Name() string {
+	return "RmnHomePoller"
+}
+
+func convertOnChainConfigToRMNHomeChainConfig(
+	lggr logger.Logger,
+	versionedConfigWithDigests []rmntypes.VersionedConfigWithDigest,
+) map[cciptypes.Bytes32]rmntypes.RMNHomeConfig {
+	if len(versionedConfigWithDigests) == 0 {
+		lggr.Warnw("no on chain RMNHomeConfigs found")
+		return map[cciptypes.Bytes32]rmntypes.RMNHomeConfig{}
+	}
+
+	rmnHomeConfigs := make(map[cciptypes.Bytes32]rmntypes.RMNHomeConfig)
+	for _, versionedConfigWithDigest := range versionedConfigWithDigests {
+		// check if the versionesconfigwithdigests are set (can be empty)
+		if versionedConfigWithDigest.ConfigDigest == (cciptypes.Bytes32{}) {
```

Comment:
Maybe create a separate function for doing validations and keep this to only translate one struct to the other?